tests/acceptance/test_download_data_for_dataset.py

```diff
@@ -0,0 +1,72 @@
+import time
+
+import pytest
+import uvicorn
+
+from multiprocessing.context import Process
+from application.settings import get_settings
+
+settings = get_settings()
+
+settings.READ_DATABASE_URL = (
+    "postgresql://postgres:postgres@localhost/digital_land_test"
+)
+settings.WRITE_DATABASE_URL = (
+    "postgresql://postgres:postgres@localhost/digital_land_test"
+)
+
+from application.app import create_app  # noqa: E402
+
+app = create_app()
+
+HOST = "0.0.0.0"
+PORT = 9000
+BASE_URL = f"http://{HOST}:{PORT}"
+
+
+def run_server():
+    uvicorn.run(app, host=HOST, port=PORT)
+
+
+@pytest.fixture(scope="session")
+def server_process():
+    proc = Process(target=run_server, args=(), daemon=True)
+    proc.start()
+    time.sleep(10)
+    yield proc
+    proc.kill()
+
+
+def test_download_data_for_dataset(
+    server_process, page, add_base_entities_to_database_yield_reset
+):
+    page.goto(BASE_URL)
+
+    # Navigate to the Brownfield site dataset page.
+    page.get_by_role("link", name="Datasets", exact=True).click()
+    page.get_by_role("link", name="Geography").click()
+    page.get_by_role("link", name="Brownfield site").click()
+
+    # Check that the "CSV" download link is correct.
+    csv_href = page.get_by_role("link", name="CSV", exact=True).first.get_attribute(
+        "href"
+    )
+
+    assert "brownfield-site" in csv_href
+    assert ".csv" in csv_href
```

Comment:
couldn't think of a better way to test this other than just checking the download link href

I did try updating the .env.example to make sure the actual dataset file is downloaded but I don't think this env is used in the testing environment causing the test to fail

Let me know if you think this could be tested in a better way and ill implement it

Reply:
I guess you could do an HTTP GET against the provided URL. 200 = good, otherwise not so good?